site/en/dev-best-practice.md

```diff
@@ -0,0 +1,145 @@
+
+### Branches and Pull Requests
+
+As mentioned in the [section on Pull Requests](#pull-requests), you should create *feature branches* that contain a narrow set of commits.
+
+### Unit tests
+
+Unit tests are code that make assertions over the behavior of the components of your main code.
+
+### Code Coverage
+
+The coverage of your tests is the percentage of lines of code that were executed running all your tests. You should strive for 100% coverage, so that everything your code does is encountered by the tests at least once. Even though a high code coverage is no panacea against bugs, it substantially reduces the amount of regressions, i.e. code changes that unexpectedly break existing behavior.
+
+### Test data
+
+The OCR-D coordination project maintains a central repository [OCR-D/assets](https://github.com/OCR-D/assets) that contains test data for unit tests. The test data consists of [unpacked OCRD-ZIP](https://ocr-d.de/en/spec/ocrd_zip) of workspaces, i.e. one or more METS XML files, as well as PAGE-XML, ALTO, images. While they use the same mechanism, **these folders are not Ground Truth**. Instead, they serve as examples of a wide range of "real-life" data, warts and all. When testing your software, consider using the data from [OCR-D/assets](https://github.com/OCR-D/assets). If none of the provided OCRD-ZIP satisfy your needs, please [open an issue in OCR-D/assets](https://github.com/OCR-D/assets/issues/new) and describe your test data needs, so we can cooperate on creating test data.
+
+Making uniform and well-described test data available to all serves everybody and the coordination project will continue to maintain and update the test data as requirements shift or specifications change. Such a test-specific corpus also improves comparability of processor performance and frees test engineers to focus on testing the actual software, not wrangling with their own test data.
+
+### Continuous Integration
+
+Continuous Integration (CI) platforms offer computing ressources to software developers that allow building and testing software. These platforms, such as [Travis CI](https://travis.com), [Circle CI](https://circleci.com), [Scrutinizer](https://scrutinizer.com) or [GitHub Actions](https://github.com/actions) can be integrated into GitHub repositories to automatically build/test when certain actions happen, such as "a PR was opened" or "new commit to master branch". The big advantage of CI is that you can automate testing of changes to your software in a variety of environments, so you can be confident that a PR did not introduce introduce regressions or break for an operating system you didn't anticipate.
```

Comment:
```suggestion
Continuous Integration (CI) platforms offer computing ressources to software developers that allow building and testing software. These platforms, such as [Travis CI](https://travis.com), [Circle CI](https://circleci.com), [Scrutinizer](https://scrutinizer.com) or [GitHub Actions](https://github.com/actions) can be integrated into GitHub repositories to automatically build/test when certain actions happen, such as "a PR was opened" or "new commit to master branch". The big advantage of CI is that you can automate testing of changes to your software in a variety of environments, so you can be confident that a PR did not introduce regressions, or make the module break for an operating system you didn't anticipate.
```

Reply:
Can't spot the change?

Reply:
Phew, GH showed the within-line changes alright in the preview, strange!

I think it was only `introduce introduce`